### ISSUE TITLE
Pass --warnings into mozlint

### ISSUE THREAD

**Issue by Pike** [mozilla/release-services#2137]
We added `--warnings` to mozlint in #1447, but that got dropped again with the move to try.

Is this the right tracker to ask for it to be added back?

CC @flodolo, @ahal

**Comment by ahal** [comment]
Yeah, this is a little tricky because now reviewbot is just running the normal lint tasks. But we need to make sure that the normal lint tasks *only* display warnings if they were scheduled by reviewbot (i.e we don't want warnings causing backouts or even showing orange on user-generated try pushes).

Maybe the reviewbot can set an environment variable that mozlint could look for to enable warnings.

**Comment by ahal** [comment]
I think another prerequisite here is re-implementing the two-pass algorithm in mozlint (i.e run the linters both with and without the changes applied so we can suppress warnings that weren't caused by the push).

**Comment by La0** [comment]
We would need to add an environment variable set by taskgraph into each analyzer tasks (like `CODE_REVIEW=1`): i'm not sure how to do that yet.

Mozlint would then do something like:
```python
if os.environ.get('CODE_REVIEW') == '1':
    flags += ['--warnings']
```

**Comment by Pike** [comment]
I was wondering if we could create separate tasks for phabricator, and run the no-warnings linters on try as usual.

Maybe even give them a dedicated platform so that they're easier to tell apart on treeherder?

**Comment by ahal** [comment]
If the reviewbot is using `mach try` (or even just `try_task_config.json`) to schedule the tasks, you can pass in `--env FOO=BAR`.

**Comment by La0** [comment]
We are pushing directly on `hg.mozilla.org/try`, but with a custom  `try_task_config.json`... So it seems that's an easy patch !

@sylvestre Are you ok with this ? Every task would then have `CODE_REVIEW=1`

**Comment by sylvestre** [comment]
I think it makes indeed sense but I would like to see the results on staging or shadow mode before going live

**Comment by ahal** [comment]
I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1555721 to implement the two-pass mode in mozlint (I think that should probably block this).
